### PR TITLE
feat: Add query parms to whats-new get request

### PR DESCRIPTION
**Pull request by TDP17**
## Closes: #3 
- Added query parameters to whats-new request
- Sent them to client 
- Validated input for type x.x (something like 1.54 will fail)

## Screenshots
Correct request 
image

Incorrect request
image

## Further questions

- Current request format http://localhost:3080/whats-new/x.x/x.x 
Do you want me to change this to http://localhost:3080/whats-new/from/1.4/to/2.50

- Should I create views for this? 

- Currently it is just accepting single digits before and after decimal ( 1.5 )
Should the request allow multiple digits after decimal. eg - something like 1.544 or 1.54? 

TIA 😄

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -28,11 +28,37 @@ module.exports = function buildApi(router) {
 			res.send(html);
 		});
 	});
-	
-	router.get('/whats-new', (req, res) => {
-		res.send('Whats up!');
+
+	router.get('/whats-new/:from/:to', (req, res) => {
+		//Function to check if string is numeric (should cover undefined, "", " ", null, true, false cases)
+		const isNumeric = (input) => {
```
Comment (flowwishthebest) [inline lib/api.js:34]: let's move this func to separate file for testing purpose

File: lib/api.js [hunk 2]
```diff
@@ -28,11 +28,37 @@ module.exports = function buildApi(router) {
...
+			return (input - 0) == input && ('' + input).trim().length > 0;
+		}
+
+		//Function to check if given query param is valid (both parts are numeric strings and number after decimal point is single digit numeric string)
+		const isValid = (string) => {
```
Comment (flowwishthebest) [inline lib/api.js:39]: let's move this func to separate file for testing purpose too

File: lib/api.js [hunk 3]
```diff
@@ -28,11 +28,37 @@ module.exports = function buildApi(router) {
...
+		//Access query params
+		const from = req.params.from;
+		const to = req.params.to;
+
+		if (isValid(from) && isValid(to)) {
+			res.status(200).send(`Whats new from ${from} to ${to}`);
+		}
+		else {
+			// TODO: If format isnt matching - change below
+			res.status(400).send("Format didn't match");
```
Comment (flowwishthebest) [inline lib/api.js:57]: In this case, we need to be as patient as possible with incorrect arguments. If incorrect params present we will send generic whats-new page with 200 OK status

File: lib/api.js [hunk 4]
```diff
@@ -28,11 +28,37 @@ module.exports = function buildApi(router) {
 			res.send(html);
 		});
 	});
-	
-	router.get('/whats-new', (req, res) => {
-		res.send('Whats up!');
+
+	router.get('/whats-new/:from/:to', (req, res) => {
```
Comment (flowwishthebest) [inline lib/api.js:32]: I think use query params is more flex then url params

**Comment by TDP17** [comment]
@flowwishthebest I made some changes, please go through them at your own convenience and let me know if there are any changes.

- I also saw that eslint is throwing errors in the form of found 4 spaces instead of tabs and ran a eslint --fix on lib/isValid.js. Will that cause problems?

- I also don't fully understand how testing is done so I kept isValid and isNumeric in the same file and exported only isValid from it (changed from es6 module syntax to commonJS syntax aswell)

**Pull request merged**